server/src/main/java/io/deephaven/server/console/completer/JavaAutoCompleteObserver.java

```diff
@@ -93,56 +82,72 @@ public void onNext(AutoCompleteRequest value) {
+                        });
+            }
         }
     }
 
-    private void getCompletionItems(GetCompletionItemsRequest request,
-            SessionState.ExportObject<ScriptSession> exportedConsole, CompletionParser parser,
+    private void handleAutocompleteRequest(AutoCompleteRequest request,
+            SessionState.ExportObject<ScriptSession> exportedConsole,
             StreamObserver<AutoCompleteResponse> responseObserver) {
-        final ScriptSession scriptSession = exportedConsole.get();
+        // Maintain compatibility with older clients
+        // The only autocomplete type supported before the consoleId was moved to the parent request was
+        // GetCompletionItems
+        final int requestId =
+                request.getRequestId() > 0 ? request.getRequestId() : request.getGetCompletionItems().getRequestId();
         try {
-            final VariableProvider vars = scriptSession.getVariableProvider();
-            final VersionedTextDocumentIdentifier doc = request.getTextDocument();
-            final CompletionLookups h = CompletionLookups.preload(scriptSession);
-            // The only stateful part of a completer is the CompletionLookups, which are already
-            // once-per-session-cached
-            // so, we'll just create a new completer for each request. No need to hang onto these guys.
-            final ChunkerCompleter completer = new ChunkerCompleter(log, vars, h);
-
-            final ParsedDocument parsed;
-            try {
-                parsed = parser.finish(doc.getUri());
-            } catch (CompletionCancelled exception) {
-                if (log.isTraceEnabled()) {
-                    log.trace().append("Completion canceled").append(exception).endl();
+            AutoCompleteResponse.Builder response = AutoCompleteResponse.newBuilder();
+
+            switch (request.getRequestCase()) {
+                case GET_COMPLETION_ITEMS: {
+                    response.setCompletionItems(getCompletionItems(request.getGetCompletionItems(), exportedConsole,
+                            parser, responseObserver));
+                    break;
+                }
+                case GET_SIGNATURE_HELP: {
+                    // Not implemented. Return empty signature list
+                    response.setSignatures(GetSignatureHelpResponse.getDefaultInstance());
+                    break;
+                }
+                case GET_HOVER: {
+                    // Not implemented. Return empty hover help
+                    response.setHover(
+                            GetHoverResponse.newBuilder().setContents(MarkupContent.getDefaultInstance()).build());
+                    break;
+                }
+                case GET_DIAGNOSTIC: {
+                    // Not implemented. Return empty diagnostics
+                    response.setDiagnostic(GetPullDiagnosticResponse.getDefaultInstance());
+                    break;
```

Comment:
It looks like the pattern here is to return success if the server knows about the type, but doesn't implement it. That said,  I wonder if it's better to *not* set any response on the GET_SIGNATURE_HELP / GET_HOVER / GET_DIAGNOSTIC cases. 

The way the implementation works now, for example, on a GET_DIAGNOSTIC request, the client needs to know that an empty `kind` field means it's not implemented?

```
message GetPullDiagnosticResponse {
    string kind = 1;
    optional string result_id = 2;
    repeated Diagnostic items = 3;
}
```

Note: as the server is previously implemented, and implemented now, a newer client won't be able to talk to an older server (REQUEST_NOT_SET will throw an exception, which seems acceptable as long as the client knows how to properly handle it).

Reply:
I think if we ever see the `default` case (in the way the code is currently structured), we should throw an illegal argument exception.

Reply:
The LSP spec is to first negotiate what the client and server support. Then the client should not send requests or expect results if the server does not support the request type in that negotiation. The client can also tell the server what it supports (like markdown or plaintext only). For now we are just assuming the web IDE which supports markdown

We could instead add a `NOT_SUPPORTED` response type to be more explicit in these cases. In that case we probably wouldn't throw in the `default` and instead just return a `NOT_SUPPORTED` response

On the web client I'm checking if the JS API has either of the new requests before registering the provider, so the web IDE won't try to send requests unless the JS API has the method we expect.